akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -100,7 +98,7 @@ private[http] object OutgoingConnectionBlueprint {
         .mapConcat(ConstantFun.scalaIdentityFunction)
         .via(new PrepareResponse(parserSettings))
 
-      val terminationFanout = b.add(Broadcast[HttpResponse](2))
+      val terminationFanout = b.add(Broadcast[HttpResponse](2, eagerCancel = true))
```

Comment:
I just checked, it's not enough - the changes below and this change make the specific test pass.
I will look deeper into the semantics to see what to do with the changes below.